File: src/test/java/com/sanderverbruggen/kotlinblogexamples/java/PersonTest.java
package com.sanderverbruggen.kotlinblogexamples.java;

import org.junit.jupiter.api.Test;

import java.time.LocalDate;
import java.time.Month;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Testing the Java class Person in Java
 */
class PersonTest {

    private final Person person = new Person("Sander", LocalDate.of(2000, Month.JANUARY, 1), "sander");

    @Test
    void testPersonAgeNow() {
        assertThat(person.age()).isGreaterThan(20);
    }

    @Test
    void testPersonAgeReferenceDate() {
        assertThat(person.age(LocalDate.of(2003, Month.MARCH, 3))).isEqualTo(3);
    }
}

File: src/test/kotlin/com/sanderverbruggen/kotlinblogexamples/kotlin/PersonTest.kt
package com.sanderverbruggen.kotlinblogexamples.kotlin

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDate
import java.time.Month

class PersonTest {
    val person = Person("Sander", LocalDate.of(2000, Month.JANUARY, 1), "sander")

    @Test
    internal fun `test person age now`() {
        assertThat(person.age()).isGreaterThan(20)
    }

    @Test
    internal fun `test person age on reference date`() {
        assertThat(person.age(LocalDate.of(2003, Month.MARCH, 3))).isEqualTo(3)
    }
}
